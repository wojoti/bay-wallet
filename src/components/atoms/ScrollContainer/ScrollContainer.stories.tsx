import {ComponentMeta, ComponentStory} from '@storybook/react-native';
import {ScrollContainer} from 'components/atoms';
import {Text, View} from 'react-native';
import {ScrollContainerProps} from './ScrollContainer';

export default {
  title: 'Atoms/ScrollContainer',
  component: ScrollContainer,
  decorators: [
    Story => (
      <View style={{borderWidth: 1}}>
        <Story />
      </View>
    ),
  ],
  argTypes: {
    justify: {
      control: {type: 'select'},
      options: [
        'flex-start',
        'flex-end',
        'center',
        'space-between',
        'space-around',
        'space-evenly',
      ],
    },
    align: {
      control: {type: 'select'},
      options: ['flex-start', 'flex-end', 'center', 'stretch', 'baseline'],
    },
    background: {
      control: {type: 'color'},
    },
  },
} as ComponentMeta<typeof ScrollContainer>;
const Template: ComponentStory<typeof ScrollContainer> = (
  args: ScrollContainerProps,
) => (
  <ScrollContainer {...args}>
    <Text>Example text</Text>
    <Text>Example text</Text>
    <Text>Example text</Text>

    <Text>Example text</Text>
    <Text>Example text</Text>
    <Text>Example text</Text>

    <Text>Example text</Text>
    <Text>Example text</Text>
    <Text>Example text</Text>

    <Text>Example text</Text>
    <Text>Example text</Text>
    <Text>Example text</Text>

    <Text>Example text</Text>
    <Text>Example text</Text>
    <Text>Example text</Text>

    <Text>Example text</Text>
    <Text>Example text</Text>
    <Text>Example text</Text>

    <Text>Example text</Text>
    <Text>Example text</Text>
    <Text>Example text</Text>
  </ScrollContainer>
);

export const Example = Template.bind({});
Example.args = {
  testId: '',
  align: 'flex-start',
  justify: 'flex-start',
  flex: 0,
};
