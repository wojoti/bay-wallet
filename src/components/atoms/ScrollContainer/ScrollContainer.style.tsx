import {StyleSheet} from 'react-native';

interface Props {
  justify?:
    | 'flex-start'
    | 'flex-end'
    | 'center'
    | 'space-between'
    | 'space-around'
    | 'space-evenly';
  align?: 'flex-start' | 'flex-end' | 'center' | 'stretch' | 'baseline';
  flex?: number;
  direction?: 'row' | 'column';
}
const style = ({justify, align, flex, direction}: Props) =>
  StyleSheet.create({
    wrapper: {
      justifyContent: justify || 'flex-start',
      alignItems: align || 'stretch',
      flex: flex,
      flexDirection: direction || 'column',
      width: '100%',
    },
  });
export default style;
