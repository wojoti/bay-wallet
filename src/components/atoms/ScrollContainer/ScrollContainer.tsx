import {PropsWithChildren} from 'react';
import {ScrollView, StyleProp, ViewStyle} from 'react-native';
import style from './ScrollContainer.style';

export type ScrollContainerProps = PropsWithChildren<{
  justify?:
    | 'flex-start'
    | 'flex-end'
    | 'center'
    | 'space-between'
    | 'space-around'
    | 'space-evenly';
  align?: 'flex-start' | 'flex-end' | 'center' | 'stretch' | 'baseline';
  flex?: number;
  testId?: string;
  direction?: 'row' | 'column';
  customStyle?: StyleProp<ViewStyle>;
}>;

const ScrollContainer = ({
  justify,
  align,
  flex,
  direction,
  testId,
  customStyle,
  children,
}: ScrollContainerProps) => {
  return (
    <ScrollView
      contentContainerStyle={[
        style({justify, align, flex, direction}).wrapper,
        customStyle,
      ]}
      testID={testId}>
      {children}
    </ScrollView>
  );
};

export default ScrollContainer;
