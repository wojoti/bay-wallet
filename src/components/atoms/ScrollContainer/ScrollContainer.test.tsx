import '@testing-library/jest-native/extend-expect';
import {render, screen} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import ScrollContainer, {ScrollContainerProps} from './ScrollContainer';

const props: ScrollContainerProps = {
  testId: 'test-scrollcontainer-id',
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(<ScrollContainer {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render ScrollContainer', () => {
  render(<ScrollContainer {...props} />);
  const ScrollContainerElement = screen.getByTestId('test-scrollcontainer-id');
  expect(ScrollContainerElement).toBeOnTheScreen();
});
