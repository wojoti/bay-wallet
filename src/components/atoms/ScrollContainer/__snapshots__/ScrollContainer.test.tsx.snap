// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<RCTScrollView
  contentContainerStyle={
    [
      {
        "alignItems": "stretch",
        "flex": undefined,
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "width": "100%",
      },
      undefined,
    ]
  }
  testID="test-scrollcontainer-id"
>
  <View />
</RCTScrollView>
`;
